### PR TITLE
Subscriber Rate Limiting Error Handling

### PR DESCRIPTION
**Pull request by karmanyaahm**
As discussed in #unifiedpush-devs:matrix.org

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -661,7 +661,7 @@ func (s *Server) handlePublish(w http.ResponseWriter, r *http.Request, v *visito
 func (s *Server) handlePublishMatrix(w http.ResponseWriter, r *http.Request, v *visitor) error {
 	_, err := s.handlePublishWithoutResponse(r, v)
 	if err != nil {
-		return &errMatrix{pushKey: r.Header.Get(matrixPushKeyHeader), err: err}
+		return err
```
Comment (karmanyaahm) [inline server/server.go:664]: Already wrapped in `transformMatrixJSON`, so no need to double wrap.

**Review by karmanyaahm — COMMENTED** [review]
1. Fixed Matrix error wrapping and pushkeys
2. Subscribe to topic before running UP tests
3. Add a Matrix test for 507 case
4. Change a Matrix test to avoid rejecting for rate limits

**Comment by karmanyaahm** [comment]
Also, if you really don't like 507, any other 5xx code should do. We can even make up our own new code.

**Comment by binwiederhier** [comment]
Thanks

**Pull request merged**